### DISCUSSION TITLE
Initial value of radio buttons isn’t used in calculation

### DISCUSSION THREAD

**Discussion by stefexitable** [solspace/craft-freeform#1672] [Questions & Answers] [answered]
I set up a donation form using the instructions on the [support page](https://docs.solspace.com/craft/freeform/v5/guides/speciality-forms/stripe-donation-payment/#instructions).

I created a list of checkboxes for amount, a custom amount field and a calculated field. The checkbox has a default selected option, although the value of the default selected option is not set when the page is loaded. Therefor the calculated amount doesn’t work. And so the Stripe plugin doesn’t load. When I click a radio, the value is set and the calculation is correct. Although the Stripe plugin still isn’t loaded. 

When I don’t set a default amount, the calculation isn’t working either.

Some other things I’ve tried:
- Use select instead of radio. Initial value also isn’t set.
- Use the value of the radio/select directly in Stripe as dynamic payment. The initial value isn’t set, and Stripe doesn’t load.

![Screenshot 2024-12-06 at 11 28 30](https://github.com/user-attachments/assets/1445e991-3dc5-4ac2-ac26-fba2d9b5e049)
![Screenshot 2024-12-06 at 11 31 09](https://github.com/user-attachments/assets/530f5b12-e9ee-412a-8811-5ad00602ca1d)

### How can we reproduce this?

1. Create a donation form following these instructions: https://docs.solspace.com/craft/freeform/v5/guides/speciality-forms/stripe-donation-payment/#instructions
2. Open the donation form

### Freeform Edition

Pro

### Freeform Version

5.7.2

### Craft Version

4.13.1.1

### When did this issue start?

Fresh install of Freeform

**Comment by kjmartens** [comment]
Sorry for the trouble @stefexitable,

You should have the Stripe Payment field on Page 2, however. This is important for the reasons listed here:
https://docs.solspace.com/craft/freeform/v5/integrations/stripe-payments/#place-stripe-field-on-last-page

However, I can see that without any action on the Radios field IF it's defaulted to a value (e.g. defaults to `5` and the user does not change it to something else), you will likely end up with a Calculation value that is not correct, even if you place the Stripe Payments field on a second page. This is why the guide uses a Dropdown with a "Select..." option. You could set NO default for the Radios, forcing them to select an option.

That would be the workaround to get this working for you right now.

In the meantime, I will see if there's a way for us to make the Calculation be able to recalculate after the form is loaded with defaults, etc, or something like that. It would resolve the issue of using an Amount field with a default value selected (and a user not changing it). 🙂

**Reply by stefexitable** [reply]
Thanks for your response @kjmartens . In my initial setup I indeed placed it on the last page, although it didn’t work either. For faster debugging, I added it to the first page. Although for this ticket, it might have been better if I moved it back to the last page.

Anyway, thanks for the workaround, and for looking into enhanced options for the calculation!

**Comment by kjmartens** [comment, marked as answer]
This is now resolved in Freeform 5.8.3+ 🙂

**Reply by stefexitable** [reply]
That’s great news, thanks!
